api/files/views.py

```diff
@@ -35,7 +35,249 @@ def get_file(self, check_permissions=True):
 
 
 class FileDetail(generics.RetrieveUpdateAPIView, FileMixin):
-    """Details about a specific file.
+    """Details about files and folders. *Writeable*.
+
+    Welcome to the Files API.  Brace yourself, things are about to get *weird*.
+
+    The Files API is the one place in the OSF API where we break hard from the JSON-API spec.  This is because most of
+    the behind-the-scenes moving, uploading, deleting, etc. of files and folders is actually be handled for us by a
+    nifty piece of software called [WaterButler](https://github.com/CenterForOpenScience/waterbutler).  WaterButler lets
+    us interact with files stored on different cloud storage platforms through a consistent API.  However, it uses
+    different conventions for requests, responses, and URL-building, so pay close attention to the documentation for
+    [actions](#actions).  The exceptions are the "Get Info" and "Checkout" actions, which are OSF-centric.
+
+    If your file or folder is stored on a non-OSF provider, the file metadata will be retrieved if not already present
```

Comment:
You can not access information about files that have not been access via some other means first.
Accessing a file via node files will assign that file a mongoid which can then be used to look up the file at this endpoint.